### PR TITLE
Workspace: adjust `Path` when executing plugins

### PR DESCRIPTION
**Pull request by compnerd**
Package plugins depend on the plugin library, which is not located in
the `Path` environment variable by default.  Adjust the path to ensure
that the plugin is able to locate the library.

Fixes: #5656

_[One line description of your change]_

### Motivation:

_[Explain here the context, and why you're making that change. What is the problem you're trying to solve.]_

### Modifications:

_[Describe the modifications you've done.]_

### Result:

_[After your change, what will change.]_

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Review by tomerd — APPROVED** [review]

**Comment by compnerd** [comment]
Yeah, I'd like to verify it as well, but yes, if this does fully resolve the issue, we should backport to 5.7 as well.

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by compnerd** [comment]
@swift-ci please build toolchain windows platform

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
From a local test of just the executable, it seems to "work" (as in the binary starts up and fails somewhere during execution of its code).  But being able to validate that it works with an actual build would be nice.

**Comment by compnerd** [comment]
@ericsink - would you be able to test https://ci-external.swift.org/job/swift-PR-build-toolchain-windows/135/artifact/build/artifacts/installer.exe ?

**Comment by ericsink** [comment]
@compnerd Strictly speaking, that new build seems to fix the Path problem, but then I bump into the *other* problem mentioned in my original forum post:

https://forums.swift.org/t/known-case-of-a-working-build-tool-plugin-on-windows/58825

I had been hoping this other problem was somehow also related to the Path, but it appears to be distinct.  In a nutshell, it seems that the build tool plugin always dies returning 1033:

```
error: plugin process ended by an uncaught signal: 1033 <command: "C:\Users\eric\dev\swiftpm-buildtool-plugin-examples\Example 1 - SwiftGen\.build\plugins\cache\SwiftGenPlugin.exe">, <output:
>
```

Note that the build tool plugin code, as long it compiles, does not affect the error behavior:

```
@main
struct SwiftGenPlugin: BuildToolPlugin {

    func createBuildCommands(context: PluginContext, target: Target) throws -> [Command] {
        // TODO nothing I put here seems to matter
        // for example, printing something to stderr here does not get included in the error message
        return []
    }
```

Bottom line: This PR fixes one problem, but AFAICT, unless I am doing something wrong, build tool plugins still don't work on Windows at all.

**Comment by compnerd** [comment]
Well, I am a fan of having focused changes.  I think that this change accomplishes what it was meant to do.  The failure that you are seeing is interesting.  I suspect that it is something different - likely a bad expectation somewhere or argument passing is triggering this.  Quickly checking the binary that was generated, running it locally does indicate that it immediately fails as well ... though its mode of failure is different but interesting.  The error level is -1073740791.  Using 2s complement, we see that this is 0xc0000409.  That indicates that it is a STACK_BUFFER_UNDERRUN.  A quick look at the stack trace shows the following:
```
 # Child-SP          RetAddr               Call Site
00 000000dc`68eff520 00007ffd`bc7c24b1     ucrtbase!invoke_watson+0x18
01 000000dc`68eff550 00007ffd`bc7c2379     ucrtbase!_invalid_parameter+0x12d
02 000000dc`68eff590 00007ffd`bc82476f     ucrtbase!invalid_parameter_noinfo+0x9
03 000000dc`68eff5d0 00007ffd`5e712474     ucrtbase!setvbuf+0x8f
04 000000dc`68eff630 00007ffd`5f99b0e6     PackagePlugin!$s13PackagePlugin0B0PAAE4mainyyYaKFZ+0x344
05 000000dc`68eff960 00007ffd`5f99b8a7     swift_Concurrency!$ss27AsyncThrowingFilterSequenceV8IteratorV10isIncludedySb7ElementQzYaKcvg+0x2c16
06 000000dc`68eff9a0 00007ffd`5f92ab50     swift_Concurrency!swift_job_run+0x57
07 000000dc`68effa00 00007ffd`5f92a92c     dispatch!dispatch_assert_queue_barrier+0x510
08 000000dc`68effa60 00007ffd`5f9354e4     dispatch!dispatch_assert_queue_barrier+0x2ec
09 000000dc`68effad0 00007ffd`bc7c1bb2     dispatch!dispatch_prohibit_transition_to_multithreaded+0x2534
0a 000000dc`68effb90 00007ffd`be877034     ucrtbase!thread_start<unsigned int (__cdecl*)(void *),1>+0x42
0b 000000dc`68effbc0 00007ffd`befe2651     KERNEL32!BaseThreadInitThunk+0x14
0c 000000dc`68effbf0 00000000`00000000     ntdll!RtlUserThreadStart+0x21
```
A closer look at frame 3:
```
03 000000dc`68eff5d0 00007ffd`5e712474     : 00007ffd`bc88f4e8 00000000`00000000 00000000`00000040 00000000`00000000 : ucrtbase!setvbuf+0x8f
```
indicates that somehow we ended up with `setvbuf(0x7ffdbc88f4e8, NULL, 0x40, 0);` which is triggering Dr Watson ...

**Comment by compnerd** [comment]
I think that given that we know that this fixes part of the problem, let's go ahead and merge this.  I'll set up a PR for a backport to 5.7 as well.  We can continue to investigate the second issue separately as you have already filed an issue for that.
